File: ria/java/gof/src/main/java/com/gof/pattern/abstractFactory/before/WhiteShipFactory.java
package com.gof.pattern.abstractFactory.before;

import com.gof.pattern.factoryMethod.after.DefaultShipFactory;
import com.gof.pattern.factoryMethod.after.Ship;
import com.gof.pattern.factoryMethod.after.WhiteShip;

public class WhiteShipFactory extends DefaultShipFactory {

    @Override
    public Ship createShip() {
        WhiteShip ship = new WhiteShip();
        ship.setAnchor(new WhiteAnchor());
        ship.setWheel(new WhiteWheel());
        return ship;
    }
}

File: ria/gof/src/main/kotlin/com/gof/pattern/abstractFactory/before/WhiteShipFactory.kt
package com.gof.pattern.abstractFactory.before

import com.gof.pattern.factory.DefaultShipFactory
import com.gof.pattern.factory.Ship
import com.gof.pattern.factory.WhiteShip

class WhiteShipFactory : DefaultShipFactory() {

    override fun createShip(): Ship {
        val ship = WhiteShip()
        ship.anchor = WhiteAnchor()
        ship.wheel = WhiteWheel()
        return ship
    }
}
